profiling/ldms-kokkos-sampler/kp_sampler.cpp

```diff
@@ -0,0 +1,206 @@
+				printf("KokkosP: end-parallel-for:        %s\n", (endForCallee == NULL) ? "no" : "yes");
+				printf("KokkosP: end-parallel-scan:       %s\n", (endScanCallee == NULL) ? "no" : "yes");
+				printf("KokkosP: end-parallel-reduce:     %s\n", (endReduceCallee == NULL) ? "no" : "yes");
+			}
+		}
+	}
+
+	free(envBuffer);
+
+	kernelCounter = 0;
+	uniqID = 1;
+
+	const char* tool_sample_rate = getenv("KOKKOS_SAMPLER_RATE");
+	if( NULL != tool_sample_rate ) {
+		kernelSampleRate = atoi( tool_sample_rate );
+	} else {
+		kernelSampleRate = 101;
+	}
+
+	if( tool_verbosity > 0 ) {
+		printf("KokkosP: Sampling rate set to: %llu\n", tool_sample_rate );
+	}
+}
+
+extern "C" void kokkosp_finalize_library() {
+
+}
+
+extern "C" void kokkosp_begin_parallel_for(const char* name, const uint32_t devID, uint64_t* kID) {
+	*kID = uniqID++;
+
+	if( ((*kID) % kernelSampleRate ) == 0 ) {
+		if( tool_verbosity > 0 ) {
+			printf("KokkosP: sample %llu calling child-begin function...\n", *kID);
+		}
+
+		if( NULL != beginForCallee ) {
+			(*beginForCallee)(name, devID, kID);
+		}
+	}
+
+}
+
+extern "C" void kokkosp_end_parallel_for(const uint64_t kID) {
+	if( (kID % kernelSampleRate ) == 0 ) {
+		if( tool_verbosity > 0 ) {
+			printf("KokkosP: sample %llu calling child-end function...\n", kID);
+		}
+
+		if( NULL != endForCallee ) {
+			(*endForCallee)(kID);
+		}
+	}
+}
+
+extern "C" void kokkosp_begin_parallel_scan(const char* name, const uint32_t devID, uint64_t* kID) {
+	*kID = uniqID++;
+
+	if( ((*kID) % kernelSampleRate ) == 0 ) {
+		if( tool_verbosity > 0 ) {
+			printf("KokkosP: sample %llu calling child-begin function...\n", *kID);
+		}
+
+		if( NULL != beginScanCallee ) {
+			(*beginScanCallee)(name, devID, kID);
+		}
+	}
+}
+
+extern "C" void kokkosp_end_parallel_scan(const uint64_t kID) {
+	if( ( kID % kernelSampleRate ) == 0 ) {
+		if( tool_verbosity > 0 ) {
+			printf("KokkosP: sample %llu calling child-end function...\n", kID);
+		}
+
+               	if( NULL != endScanCallee ) {
+                       	(*endScanCallee)(kID);
+               	}
+       	}
+}
+
+extern "C" void kokkosp_begin_parallel_reduce(const char* name, const uint32_t devID, uint64_t* kID) {
+	*kID = uniqID++;
+
+	if( ((*kID) % kernelSampleRate ) == 0 ) {
+		if( tool_verbosity > 0 ) {
+			printf("KokkosP: sample %llu calling child-begin function...\n", *kID);
+		}
+
+                if( NULL != beginReduceCallee ) {
+                        (*beginReduceCallee)(name, devID, kID);
+                }
+        }
+}
+
+extern "C" void kokkosp_end_parallel_reduce(const uint64_t kID) {
+	if( ( kID  % kernelSampleRate ) == 0 ) {
+		if( tool_verbosity > 0 ) {
+			printf("KokkosP: sample %llu calling child-end function...\n", kID);
+		}
+
+                if( NULL != endReduceCallee ) {
+                        (*endReduceCallee)(kID);
+                }
+       	}
+}
+
+extern "C" void kokkosp_push_profile_region(char* regionName) {
+
+}
+
+extern "C" void kokkosp_pop_profile_region() {
+
+}
```

Comment:
Why define them if they don't do anything?

Reply:
Those specific functions are currently in discussion regarding how best to handle them. I can delete them since they currently aren't doing anything.